### PR TITLE
docs(behavior_path_planner): update pull out docs

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

update pull_out docs related  to https://github.com/autowarefoundation/autoware.universe/pull/1438

please see
- https://github.com/tier4/autoware.universe/blob/docs/update_pull_out_docs/planning/behavior_path_planner/README.md
- https://autowarefoundation.github.io/autoware.universe/pr-1587/planning/behavior_path_planner/

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/1438
## Tests performed

not tested because changing only docs

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/README.md [hunk 1]
```diff
@@ -253,39 +253,118 @@ The Pull Out module is activated when ego-vehicle is stationary and footprint of
...
+![pull_out_collision_check](./image/pull_out_collision_check.drawio.svg)
 
 #### **Path Generation**
 
-The path is generated with a certain margin from the left edge of the shoulder lane.
+There are two path generation methods.
+
+##### **shift pull out**
+
+Pull out distance is calculated by the speed, lateral deviation, and the lateral jerk. The lateral jerk is searched for among the predetermined minimum and maximum values, and the one satisfies ready conditions described above is output.
```
Comment (TakaHoribe) [inline planning/behavior_path_planner/README.md:296]:
```suggestion
Pull out distance is calculated by the speed, lateral deviation, and the lateral jerk. The lateral jerk is searched for among the predetermined minimum and maximum values, and the one that satisfies the ready conditions described above is selected.
```

It is said, "It is always ready if the conditions of the request are met" above. What do you mean "satisfies ready conditions described above"?

File: planning/behavior_path_planner/README.md [hunk 2]
```diff
@@ -253,39 +253,118 @@ The Pull Out module is activated when ego-vehicle is stationary and footprint of
...
+| pull_out_sampling_num             | -      | int    | Number of samplings in the minimum to maximum range of lateral_jerk                                                  | 4             |
+| before_pull_out_straight_distance | [m]    | double | distance before pull out start point                                                                                 | 5.0           |
+| maximum_lateral_jerk              | [m/s3] | double | maximum lateral jerk                                                                                                 | 2.0           |
+| minimum_lateral_jerk              | [m/s3] | double | minimum lateral jerk                                                                                                 | 0.5           |
+| minimum_shift_pull_out_distance   | [m]    | double | minimum shift pull out distance. if calculated pull out distance is shorter than this, use this for path generation. | 20.0          |
+
+##### **geometric pull out**
+
+Generate two arc paths with discontinuous curvature. ego-vehicle stops once in the middle of the path to control the steer on the spot.
```
Comment (TakaHoribe) [inline planning/behavior_path_planner/README.md:320]:
```suggestion
Generate two arc paths with discontinuous curvature. Ego-vehicle stops once in the middle of the path to control the steer on the spot.
```

Comment (kosuke55) [reply planning/behavior_path_planner/README.md]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/1587/commits/3fe45f3a0539707e19b52304e43c03d3a05e6092

File: planning/behavior_path_planner/README.md (hunk 1 above)
Comment (kosuke55) [reply planning/behavior_path_planner/README.md]: This is wrong, I just brought up old docs sentences. Fixed in https://github.com/autowarefoundation/autoware.universe/pull/1587/commits/3fe45f3a0539707e19b52304e43c03d3a05e6092.

Comment (kosuke55) [reply planning/behavior_path_planner/README.md]: "It is always ready if the conditions of the request are met" is true.

**Review by TakaHoribe — APPROVED** [review]
LGTM

**Comment by kosuke55** [comment]
- [ ] write the reason why vehicle_info's max_steer is not used.

**Pull request merged**